azurelinuxagent/ga/policy/regorus.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+import json
+import subprocess
+import os
+import tempfile
+
+
+def get_regorus_path():
+    """
+    Returns path to Regorus executable. The executable is not yet officially released as part of the agent package.
+    After release, executable will be placed at /var/lib/waagent/WALinuxAgent-x.x.x.x/bin/regorus.
+    Currently, the executable is copied into the agent directory for unit testing, and this method is mocked.
+    """
+    regorus_exe = os.path.join(os.getcwd(), "bin", "regorus")
+    return regorus_exe
+
+
+class Engine:
+    """
+    This class implements the basic operations for the Regorus policy engine via subprocess.
+    Any errors thrown in this class should be caught and handled by PolicyEngine.
+    """
+
+    def __init__(self):
+        self._engine = None
+        self._policy_file = None
+        self._data_file = None
+        self._input_json = None
+
+    def add_policy(self, policy_path):
+        """Policy_path is expected to point to a valid Regorus file."""
+        if not os.path.exists(policy_path) or not policy_path.endswith('.rego'):
+            raise Exception("Policy path {0} is not a valid .rego file.".format(policy_path))
```

Comment:
what's in a *.rego file? JSON content? if yes, let's use ".json" instead. The fact that regorus is being used under the covers is an implementation detail and ".json" will be more familiar to users than "*.rego"